addon/ng2/blueprints/component/index.js

```diff
@@ -1,26 +1,44 @@
 var stringUtils = require('ember-cli/lib/utilities/string');
 var path = require('path');
+var process = require('process');
 
 module.exports = {
   description: '',
-
-  locals: function(options) {
-    var parsedPath = path.parse(options.entity.name);
-    // build the locals for generating the name & path
-    return {
+  
+  normalizeEntityName: function(entityName){
+    var rootPath = path.join(this.project.root, 'src', 'app');
+    
+    var outputPath = path.join(process.env.PWD, entityName);
+    
+    if (outputPath.indexOf(rootPath) < 0) {
+      throw `Invalid component path: "${entityName}" cannot be above the "${path.join('src', 'app')}" directory`; 
+    }
+    
+    var adjustedPath = outputPath.replace(rootPath, '');
+    
+    var parsedPath = path.parse(adjustedPath);
+    this.dynamicPath = {
       name: parsedPath.name,
       path: parsedPath.dir
     };
+    return parsedPath.name;
+  },
+  
+  locals: function(options){
+    return {
+      dynamicPath: this.dynamicPath.path
+    }
   },
 
   fileMapTokens: function(options) {
     // Return custom template variables here.
+    var _this = this;
     return {
       __name__: function(options) {
-        return options.locals.name;
+        return _this.dynamicPath.name;
```

Comment:
I keep forgetting that... I use them a **lot** when doing TypeScript so when I'm doing JavaScript I think I'm always restricted to ES5 syntax. Good catch, new PR pushed!